docs/site/tutorials/authentication/Quick-Authentication-Tutorial-Session-Tokens.md

```diff
@@ -0,0 +1,315 @@
+
+Now we need to tell our application that this strategy exists and can be used.
+
+Open `src/application.ts` and add the following lines:
+
+```ts
+import {
+  AuthenticationComponent,
+  registerAuthenticationStrategy,
+} from '@loopback/authentication';
+import {BasicAuthenticationStrategy} from './strategies/basic-strategy';
+
+// At the bottom of the constructor function
+
+this.component(AuthenticationComponent);
+registerAuthenticationStrategy(this, BasicAuthenticationStrategy);
+```
+
+## Start using the Basic authentication strategy in controllers
+
+To employ our authentication in controllers, we can use the `@authenticate()`
+decorator.
+
+For OAuth2 you may want to create a `/users/login` route. You could add it to
+your existing UserController (or to the PingController, if that is all you have
+right now).
+
+Add the following lines to your controller:
+
+```ts
+import { authenticate, AuthenticationBindings } from '@loopback/authentication';
+import { UserProfile } from '@loopback/security';
+
+// In the constructor's arguments
+
+    @inject(AuthenticationBindings.CURRENT_USER, { optional: true }) private user: UserProfile,
+    // If you decided earlier to pass the full User object to controllers, put `UserProfile & User`
+
+// Add the API endpoint
+
+  @authenticate('BasicStrategy')
+  @post('/users/login')
+  async login(): Promise<{token: string}> {
+    // We expect the Basic strategy will have authenticated a user.
+    // In fact if your strategy uses `throw Error` instead of `return undefined`
+    // then we can skip this check!
+    if (this.user) {
+      console.log("/users/login has received user:", this.user);
+      // TODO Generate a unique bearer token for this user and store it in the DB
+      const token = 'example_token_999';
+      return {token};
+    } else {
+      throw new Error('User was not authenticated');
+    }
+  }
+```
+
+## Add a test to ensure it works as expected
+
+Add the following to one of your acceptance tests:
+
+```ts
+it('can log in with POST /users/login', async () => {
+  const plainAuth = 'testuser:password123';
+  const encodedAuth = Buffer.from(plainAuth).toString('base64');
+  const res = await client
+    .post('/users/login')
+    .set('Authorization', 'Basic ' + encodedAuth)
+    .expect(200);
+  expect(res.body).to.containEql({token: 'example_token_999'});
+});
+```
+
+Now running `npm test` will verify that your authentication pipeline works.
+
+## Next steps
+
+Now you have seen how to integrate with Loopback's authentication framework.
+
+From here you can adapt the strategy to meet your requirements. For example,
+you could communicate with an existing service or database for authentication.
+
+Alternatively, if you want to build a complete authentication system from
+scratch, you could follow the [JSON WebTokens
+tutorial](Authentication-Tutorial.md).
+
+Or if you want to complete the session token based system, read on.
+
+## Completing the session tokens system
```

Comment:
by session tokens , do you mean, a client web session with cookies ?

Reply:
@deepakrkris Yes, more or less.

My personal experience has been passing the tokens through the `Authorization` header, but I think using cookies would be fairly similar.